### DISCUSSION TITLE
Searchbar functionality submit button not working

### DISCUSSION THREAD

**Discussion by StefanieLoeser** [neuefische/questions#262] [Web]
Hi everybody, 

I am trying to implement a search functionality – but already the submit button is not working and I can't find the issue. Trying to log out the input value doesn't show anything in the console. 

And then also I am not quite sure how to tackle the search itself as I am working with that global state and would want that this also can be updated (bookmarks, add videos) within the search results.

Would be amazing if someone could help me – maybe find whats wrong with my  form and lead me in the right direction with the filtering process and how to update my state with the filtered data. 

[Link to PR](https://github.com/StefanieLoeser/capstone-project/pull/24)

**Comment by NikRoe** [comment]
Hey Stefanie,

your event handler should be assigned to the form itself and not be part of the button.
Otherwise, your `handleSubmit()` is unable to recognise the `event.target.searchRecord.value`.

Moreover, you are not working with `event.preventDefault()` in your `handleSubmit()` this will lead to your form being sent and is most likely not what you intend to do.

Applying those two changes should lead to the desired results. As for your following question regarding your state: I would start implementing the functionality to the search bar itself and once everything works as intended, continue working on the next challenge. Feel free to keep us in the loop regarding this.

Best,
Niklas

**Reply by StefanieLoeser** [reply]
Hi @NikRoe , thank you for your suggestion! 
I made the changes but unfortunately they didn't help. Do you have another idea?
Thank you! 
Best, Stefanie
